code/modules/recycling/conveyor2.dm

```diff
@@ -125,43 +124,39 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 		return FALSE
 	return TRUE
 
-// returns true if there is at least 1 item on the belt that can be moved
-/obj/machinery/conveyor/proc/check_belt_contents()
-	for(var/atom/movable/AM in loc)
-		if(!AM.anchored)
-			return TRUE
-	return FALSE
+// machine process
+// move items to the target location
+/obj/machinery/conveyor/process()
+	if(machine_stat & (BROKEN | NOPOWER))
+		return
+
+	//If the conveyor is broken or already moving items
+	if(!operating || conveying)
+		return
 
-// populates `affecting` with the first 30 items in the belt's contents
-/obj/machinery/conveyor/proc/get_belt_contents()
+	use_power(6)
+
+	//get the first 30 items in contents
 	affecting = list()
 	var/i = 0
-	for(var/atom/movable/AM in loc)
-		if(AM.anchored) // If it can't be moved, ignore it
+	for(var/item in loc.contents)
+		if(item == src)
 			continue
-		if(++i > MAX_CONVEYOR_ITEMS_MOVE)
+		i++ // we're sure it's a real target to move at this point
+		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
 			break
-		affecting.Add(AM)
-
-// when some item or mob moves over the conveyor belt
-/obj/machinery/conveyor/proc/conveyorCrossed(datum/source, atom/movable/AM, atom/oldLoc)
-	// if the conveyor isn't switched on or is broken / has no power
-	if(!operating || machine_stat & (BROKEN | NOPOWER))
-		return
+		affecting.Add(item)
```

Comment:
As i think, and that already written upper

`conveyor Crossed` in trigger system must be used to dirrectly populate contents of `affecting` instead of call get_belt_contents() that parce all contents of our turf EVERY TIME when something step on conveyor.
And it create new timer for EVERY THING that step on conveyor.

OR

Make system for individual conveyor item transferring.
`conveyor Crossed` create timer that try move only triger thing after 1 tick as it cross in if thing still in our turf.

Reply:
I think you may be looking at older code. I recently reverted back to using the old way of moving stuff for belts, which is using `process()`. I'm not doing signal based stuff anymore for the conveyor because to get it working like the old system, it ended up being more expensive than how the belts currently handle it.

Reply:
I read this.
I write what i think can make trigger system less expensive